driver/csiplugin/controllerserver.go

```diff
@@ -674,15 +740,37 @@ func (cs *ScaleControllerServer) CreateVolume(ctx context.Context, req *csi.Crea
 		return nil, err
 	}
 
+	// Block creating a cache volume from another volume (clone) or
+	// from a snapshot (restore)
+	if scaleVol.VolumeType == cacheVolume && (isSnapSource || isVolSource) {
+		return nil, status.Error(codes.InvalidArgument, "Creating a cache volume from another volume or snapshot is not supported")
+	}
+
 	isShallowCopyVolume := false
 	for _, reqCap := range reqCapabilities {
 		if reqCap.GetAccessMode().GetMode() == csi.VolumeCapability_AccessMode_MULTI_NODE_READER_ONLY {
 			if isSnapSource {
 				klog.Infof("[%s] Requested pvc is a shallow copy volume", loggerId)
 				isShallowCopyVolume = true
+			} else if scaleVol.VolumeType == cacheVolume {
+				if scaleVol.CacheMode == "" {
+					// cacheMode is not specified, use AFM mode RO by default for volume access mode ROX
+					scaleVol.CacheMode = afmModeRO
+				} else if scaleVol.CacheMode != afmModeRO {
+					return nil, status.Error(codes.InvalidArgument, "The volume access mode ReadOnlyMany is only supported with the AFM mode readonly (RO)")
+				}
 			} else {
 				return nil, status.Error(codes.Unimplemented, "Volume source with Access Mode ReadOnlyMany is not supported")
 			}
+		} else {
+			if scaleVol.VolumeType == cacheVolume {
+				if scaleVol.CacheMode == "" {
+					// cacheMode is not specified, use AFM mode IW by default for other volume access modes
+					scaleVol.CacheMode = afmModeIW
+				} else if scaleVol.CacheMode == afmModeRO {
+					return nil, status.Error(codes.InvalidArgument, "The AFM mode readonly (RO) is only supported with the volume access mode ReadOnlyMany")
```

Comment:
we should say cacheMode ReadOnly instead of AFM mode readonly as these value refer to storageClass param

Reply:
Done